Userland/Services/WindowServer/Window.cpp

```diff
@@ -1060,9 +1065,31 @@ void Window::set_modified(bool modified)
 String Window::computed_title() const
 {
     String title = m_title.replace("[*]"sv, is_modified() ? " (*)"sv : ""sv, ReplaceMode::FirstOnly);
+    if (m_title_username.has_value())
+        title = String::formatted("{} [{}]", title, m_title_username.value());
     if (client() && client()->is_unresponsive())
         return String::formatted("{} (Not responding)", title);
     return title;
 }
 
+ErrorOr<Optional<String>> Window::compute_title_username(ConnectionFromClient* client)
+{
+    if (!client)
+        return Error::from_string_literal("Tried to compute title username without a client");
+    auto stats = Core::ProcessStatisticsReader::get_all(true);
+    if (!stats.has_value())
+        return Error::from_string_literal("Failed to get all process statistics");
+    pid_t client_pid = TRY(client->socket().peer_pid());
```

Comment:
There's no need to check /sys/kernel/processes for this piece of data. `uid` is contained in `struct ucred` on ... every system we build LocalSocket for except macOS. 

It looks like we can use the same LOCAL_PEERCRED as FREEBSD to get the uid on that platform though.

Reply:
eh. except we still need to use it to get the uid of the root session id. All right, +1 to the yakstack on that one then. We really need a better way to get root session id info.